### PR TITLE
Find minimum usecase and Read data use case

### PR DESCRIPTION
**Pull request by Het95**

### CODE REVIEW

File: metricstics/src/controller/input_controller.py [hunk 1]
```diff
@@ -27,6 +27,28 @@ def clear_data(self):
         """Clear the data set to empty."""
         self.data = set()
 
+    def read_data(self):
+        """Create a list of calculations and perform them.
```
Comment (MajorTreble) [inline metricstics/src/controller/input_controller.py:31]: Change the comment

File: metricstics/src/model/calculation.py [hunk 2]
```diff
@@ -5,6 +5,45 @@
...
+            result["Minimum"] = 0
+            return
+
+        for i in data:
+            if i < minimum:
+                minimum = data
+
+        result["Minimum"] = minimum
+
+    def read_data(self,data,result):
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:30]: This isn't a calculation. Make a new class

File: metricstics/src/controller/input_controller.py [hunk 3]
```diff
@@ -27,6 +27,28 @@ def clear_data(self):
...
+
+                Returns:
+                    result(dict): The results of the calculations
+                """
+        result = {}
+        readdata = Calculation()
+        readdata.read_data(self.data,result)
+        return result
```
Comment (MajorTreble) [inline metricstics/src/controller/input_controller.py:39]: There is no return value

File: metricstics/src/model/calculation.py [hunk 4]
```diff
@@ -5,6 +5,45 @@
...
 
+    def calculate_minimum(self,data,result):
+        """
+                Perform minimum calculation.
+
+                Arg:
+                    data(list): the data to work with
+                    result(dic): store the result
+                """
+        data = [5,41,28,9,1]
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:16]: Don't change the data here

File: metricstics/src/model/calculation.py [hunk 5]
```diff
@@ -5,6 +5,45 @@
...
+        """
+                        Perform read data operation.
+
+                        Arg:
+                            data(list): the data to work with
+                            result(dic): store the result
+                        """
+        size = len(data)
+        if size == 0:
+            result["readdata"] = 0
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:40]: Don't add this to the dictionary

File: metricstics/src/model/calculation.py [hunk 6]
```diff
@@ -5,6 +5,45 @@
...
+    def read_data(self,data,result):
+        """
+                        Perform read data operation.
+
+                        Arg:
+                            data(list): the data to work with
+                            result(dic): store the result
+                        """
+        size = len(data)
+        if size == 0:
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:39]: You are loading data, so it will be 0

File: metricstics/src/model/calculation.py [hunk 7]
```diff
@@ -5,6 +5,45 @@
...
+            result["readdata"] = 0
+            return
+
+        with open("myfile.txt", "r+", encoding="utf-8") as file1:
+            # Reading from a file
+            print(file1.read())
```
Comment (MajorTreble) [inline metricstics/src/model/calculation.py:45]: Store it in data

File: metricstics/src/view/view.py [hunk 8]
```diff
@@ -112,3 +112,16 @@ def calculate_standard_deviation_clicked(self):
         result = self.controller.calculate_standard_deviation()
         self.output_text.delete("1.0", "end")
         self.output_text.insert("1.0", result["StandardDeviation"])
+
+    def calculate_minimum(self):
```
Comment (MajorTreble) [inline metricstics/src/view/view.py:116]: Link this in the button

**Review by MajorTreble — CHANGES_REQUESTED** [review]

File: metricstics/src/model/calculation.py (hunk 2 above)
Comment (Het95) [reply metricstics/src/model/calculation.py:30]: But the calculation class can have method for reading data on which operations according to user choice can be performed

Comment (MajorTreble) [reply metricstics/src/model/calculation.py:30]: These are math calculations. Read and write data doesn't fit.

**Review by MajorTreble — APPROVED** [review]

**Pull request merged**